### PR TITLE
Userspace pointer usage fix

### PR DESCRIPTION
**Pull request by dakotarwilliams**
This is a fixup of #202, I squashed history and maintained authorship.

### CODE REVIEW

File: src/dattobd.c [hunk 1]
```diff
@@ -4771,15 +4771,22 @@ static void post_umount_check(int dormant_ret, long umount_ret, unsigned int idx
...
-
-	LOG_DEBUG("detected block device mount: %s -> %s : %lu", dev_name, dir_name, real_flags);
-
+	char buff_dev_name[256],buff_dir_name[256];
+	ret_dev=copy_from_user(buff_dev_name,dev_name,256);
+	ret_dir=copy_from_user(buff_dir_name,dir_name,256);
+	if(0!=ret_dev || 0!=ret_dir)
+		LOG_DEBUG("detected block device Get mount params error!");
+	else
+	LOG_DEBUG("detected block device mount: %s -> %s : 0x%x", buff_dev_name,
```
Comment (adnanshaheen) [inline src/dattobd.c]: indent this

File: src/dattobd.c [hunk 2]
```diff
@@ -4803,9 +4810,12 @@ static asmlinkage long umount_hook(char __user *name, int flags){
...
 	unsigned int idx;
-
-	LOG_DEBUG("detected block device umount: %s : %d", name, flags);
-
+	char buff_dev_name[256];
+	ret=copy_from_user(buff_dev_name,name,256);
+	if(0!=ret)
+	    LOG_DEBUG("detected block device umount error:%d", ret);
+	else
+	LOG_DEBUG("detected block device umount: %s : %d", buff_dev_name, flags);
```
Comment (adnanshaheen) [inline src/dattobd.c]: indent this

File: src/dattobd.c [hunk 3]
```diff
@@ -4771,15 +4771,22 @@ static void post_umount_check(int dormant_ret, long umount_ret, unsigned int idx
...
 	long sys_ret;
 	unsigned int idx;
 	unsigned long real_flags = flags;
 
 	//get rid of the magic value if its present
 	if((real_flags & MS_MGC_MSK) == MS_MGC_VAL) real_flags &= ~MS_MGC_MSK;
-
-	LOG_DEBUG("detected block device mount: %s -> %s : %lu", dev_name, dir_name, real_flags);
-
+	char buff_dev_name[256],buff_dir_name[256];
```
Comment (adnanshaheen) [inline src/dattobd.c]: Should use `PATH_MAX` for `buff_dir_name`

**Comment by crawfxrd** [comment]
and re:

> But, is it even safe to be logging in a syscall in the first place?

`printk()` is callable from any context [\[1\]](https://www.kernel.org/doc/html/latest/driver-api/basics.html#c.printk).

**Pull request merged**